backend/homework/2/q2/src/main/java/org/example/SentimentAnalyzer.java

```diff
@@ -0,0 +1,227 @@
+package org.example;
+import java.util.*;
+
+public class SentimentAnalyzer {
+    private static LoggingSystem ls = new LoggingSystem();
+
+    /**
+     * it detects feature have positive / negative / no opinion using getOpinionOnFeature() Function
+     * @param review review given by user
+     * @param featureSet set of feature with possible variation of feature
+     * @param posOpinionWords positive opinions words
+     * @param negOpinionWords negative opinions words
+     * @return array with 1 for feature having positive opinion
+     *                    -1 for feature having negative opinion
+     *                    0 for feature having no opinion
+     */
+    public static int[] detectProsAndCons(
+            String review,
+            String[][] featureSet,
+            String[] posOpinionWords,
+            String[] negOpinionWords
+    ) {
+        int features = featureSet.length;
+        int[] featureOpinions = new int[features]; // output
+        HashSet<String> reviewWordSet =new HashSet<>();
+        addOpinionsToSet(reviewWordSet,splitIntoWords(review));
+        for(int i = 0;i < features;i++)
```

Comment:
It's ok to name variables like i, j for local scope but try better names. Also maintain uniformity for loop somewhere and foreach inside isn't a good practice until its doing something a 2 for loop setup or a 2 foreach can't do